ctapipe/reco/sklearn.py

```diff
@@ -937,21 +965,34 @@ def __call__(self, telescope_type, table):
             train = table[train_indices]
             test = table[test_indices]
 
-            cv_prediction, truth, metrics = self.cross_validate(
-                telescope_type, train, test
-            )
-            predictions.append(
-                Table(
-                    {
-                        "cv_fold": np.full(len(truth), fold, dtype=np.uint8),
-                        "tel_type": [str(telescope_type)] * len(truth),
-                        "truth": truth,
-                        "true_energy": test["true_energy"],
-                        "true_impact_distance": test["true_impact_distance"],
-                        **cv_prediction,
-                    }
+            cv_result, metrics = self.cross_validate(telescope_type, train, test)
+            if self.output_path:
```

Comment:
This is very unreadable... introducing some intermediate variables might be better, i.e:
```
results = Table(
     ...
)
results = hstack([results, cv_predictions], joint_type='exact')
```